packages/katello/katello/hostname-change.rb

```diff
@@ -87,27 +88,52 @@ def precheck
       # Taken from https://www.safaribooksonline.com/library/view/regular-expressions-cookbook/9781449327453/ch08s15.html
       hostname_regex = /^([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}$/
       unless hostname_regex === @new_hostname
-        self.fail_with_message("#{@new_hostname} is not a valid fully qualified domain name, please use a valid FQDN and try again. " \
+        self.fail_with_message("#{@new_hostname} is not a valid fully qualified domain name. Please use a valid FQDN and try again. " \
                           "No changes have been made to your system.");
       end
 
       unless @foreman_proxy_content
         STDOUT.puts "\nChecking overall health of server"
-        self.run_cmd("hammer ping", [0], "There is a problem with the server, please check 'hammer ping'")
+        self.run_cmd("hammer ping", [0], "There is a problem with the server; please check 'hammer ping'")
         STDOUT.puts "\nChecking credentials"
-        self.hammer_cmd("capsule list", [0], "There is a problem with the credentials, please retry")
+        self.hammer_cmd("capsule list", [0], "There is a problem with the credentials; please retry")
       end
 
-      if @options[:confirm]
-        response = true
-      else
-        STDOUT.print(self.warning_message)
-        response = agree('Proceed with changing your hostname? [y/n]')
+      if should_update_dns?
+        begin
+          STDOUT.puts "\nAssembling data for DNS update"
+          @answers_dns_values = answers_dns_values
+          @assembled_nsupdate_command = assembled_nsupdate_command(@answers_dns_values)
+        rescue StandardError => e # could not reach the DNS server, or something broke while assembling the data
+          STDOUT.puts e
+          fail_with_message("Error querying local DNS server for #{@old_hostname}. Make sure the 'named' service is running, or re-run with the --skip-dns option.")
+        end
+
+        @answers_dns_values.each do |field, v|
+          next unless v.nil?
+            fail_with_message """
+      Error gathering DNS data: couldn't find value for '#{field}'.
+      Make sure /etc/foreman-installer/scenarios.d/#{@options[:scenario]}-answers.yaml
```

Comment:
Perhaps this should reuse `scenarios_path` from `KatelloUtilities::Helper`. There's already another definition that uses the full path so perhaps time for another helper to avoid duplication?

I also don't like the implication that the user should change the answers file. They should **never** manually change the answers file (which is why I want to move it to `/var/lib/foreman-installer` in a future version).

Reply:
@ekohl Ok, see if you prefer the reworded error message instead.